webpack.config.js

```diff
@@ -488,14 +488,28 @@ async function initializeModuleFederation() {
             // Other containers will be forced to use the exact versions of shared modules that the web app provides.
             makeSingletonSharedModules([
                 'react',
+                'react-beautiful-dnd',
                 'react-bootstrap',
                 'react-dom',
                 'react-intl',
                 'react-redux',
                 'react-router-dom',
             ]),
         ],
-    }));
+    };
+
+    // Desktop specific code for remote module loading
+    moduleFederationPluginOptions.exposes = {
+        './root': 'components/root',
+        './crtWatcher': 'components/threading/channel_threads/posts_channel_reset_watcher',
+        './store': 'stores/redux_store.jsx',
+        './styles': 'sass/styles.scss',
+        './registry': 'registry',
+    };
+    moduleFederationPluginOptions.filename = 'remoteEntry.js';
+    moduleFederationPluginOptions.name = 'mattermost_webapp';
```

Comment:
This is already set above, and by NPM package naming conventions, it should be use a hyphen and not an underscore

Reply:
I get this error when using a hyphen:
```
[webapp] Error: Library name base (mattermost-webapp) must be a valid identifier when using a var declaring library type. Either use a valid identifier (e. g. mattermost_webapp) or use a different library type (e. g. 'type: "global"', which assign a property on the global scope instead of declaring a variable). Common configuration options that specific library names are 'output.library[.name]', 'entry.xyz.library[.name]', 'ModuleFederationPlugin.name' and 'ModuleFederationPlugin.library[.name]'.
```

Reply:
Oh, I see. So they create a variable with the same name as the container somewhere, so it doesn't accept `mattermost-webapp` because you can't have a hyphen in a variable name. In that case, `mattermost_webapp` should be fine, but can you set it in the object above where it's currently set to `mattermost-webapp`?